contract/contracts/Tips.sol

```diff
@@ -18,45 +25,95 @@ contract Tips {
+        require(
+            pushLimit >= _teamMembersAddresses.length,
+            "You have too much team members"
+        );
+        uint totalAmount = msg.value + remainder;
+        uint256 amountPerAddress = totalAmount / _teamMembersAddresses.length;
+        
+        // Update the remainder
+        remainder= totalAmount % _teamMembersAddresses.length;
 
+        // Should fail if the contract balance is insufficient
         for (uint256 i = 0; i < _teamMembersAddresses.length; i++) {
-            bool transferSuccess = payable(_teamMembersAddresses[i]).send(amountPerAddress);
-            require(transferSuccess, "Failed to send tip.");
+            // Check for zero addresses
+            require(_teamMembersAddresses[i] != address(0), "Invalid address");
+            require(
+                address(this).balance >= amountPerAddress,
+                "Insufficient contract balance"
+            );
+            
+            (bool sent, ) = _teamMembersAddresses[i].call{value: amountPerAddress}("");
+            require(sent, "Failed to send ETH");
         }
 
         emit PushTip(msg.sender, _teamMembersAddresses, msg.value, amountPerAddress);
     }
 
-    function sendTip(address[] memory _teamMembersAddresses) positiveAmountRequired external payable {
-        uint256 amountPerAddress = msg.value / _teamMembersAddresses.length;
+    function sendTip(address [] calldata _teamMembersAddresses) positiveAmountRequired whenNotPaused external payable {
+        require(_teamMembersAddresses.length > 0, "Must have at least one team member");
+        uint totalAmount = msg.value + remainder;
+        uint256 amountPerAddress = totalAmount / _teamMembersAddresses.length;
+        
+        // Update the remainder
+        remainder= totalAmount % _teamMembersAddresses.length;
 
         for (uint256 i = 0; i < _teamMembersAddresses.length; i++) {
+            // Check for zero addresses
+            require(_teamMembersAddresses[i] != address(0), "Invalid address");
             balance[_teamMembersAddresses[i]] += amountPerAddress;
         }
 
         emit SendTip(msg.sender, _teamMembersAddresses, msg.value, amountPerAddress);
     }
 
-    function withdraw() external payable {
-        uint256 tipEarned = balance[msg.sender];
-        require(tipEarned > 0, "No tips to withdraw.");
+    // TODO: Protection for reentrancy attack
+    function withdraw() external nonReentrant whenNotPaused{
+        uint256 senderBalance = balance[msg.sender];
+        require(senderBalance > 0, "No tips to withdraw.");
 
-        bool tipSuccess = payable(msg.sender).send(tipEarned);
-        require(tipSuccess, "Failed to withdraw tips.");
+        (bool sent,) = msg.sender.call{value: senderBalance}("");
+        require(sent, "Failed to withdraw tips.");
 
         balance[msg.sender] = 0;
+        require(balance[msg.sender] == 0, "Failed to zero out balance");
 
-        emit TipWithdrawal(msg.sender, tipEarned);
```

Comment:
i just watch reentrancy attack, i think we should update the balance to 0 in above before sending eth

Reply:
Yes, but we have reentrency gard so.